### ISSUE TITLE
[return-await] wants me to await in a normal function

### ISSUE THREAD

**Issue by LinusU** [typescript-eslint/typescript-eslint#1930]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/return-await": ["error", "always"]
  }
}
```

```TS
declare function foo(): Promise<boolean>

function bar (baz: boolean): Promise<boolean> | boolean {
  if (baz) {
    return true
  } else {
    return foo() // <-- returning an awaited promise is required in this context (eslint@typescript-eslint/return-await)
  }
}
```

**Expected Result**

Not telling me to await, since I cannot await there...

**Actual Result**

It tells me to insert await

**Additional Info**

Here is my actual function:

```typescript
type PermissionContext = Partial<Pick<Context, 'dataloader' | 'user'>>

export function hasPermission (permission: AuthorizationPermission.Authenticated, context: Pick<Context, 'user'>): boolean
export function hasPermission (permission: AuthorizationPermission.RestaurantWrite, context: Pick<Context, 'dataloader' | 'user'>, restaurantId: string): Promise<boolean>
export function hasPermission (permission: AuthorizationPermission.SuperUser, context: Pick<Context, 'user'>): boolean
export function hasPermission (permission: AuthorizationPermission.UserWrite, context: Pick<Context, 'user'>, username: string): boolean

export function hasPermission (permission: AuthorizationPermission, { dataloader, user }: PermissionContext, resource?: string): Promise<boolean> | boolean {
  if (user == null) return false
  if (user.isSuperUser) return true

  switch (permission) {
    case AuthorizationPermission.Authenticated:
      return true
    case AuthorizationPermission.UserWrite:
      return user.username === resource
    case AuthorizationPermission.RestaurantWrite:
      return dataloader.accessByUsernameAndRestaurantId(user.username, resource)
    case AuthorizationPermission.SuperUser:
      return false
    default:
      unreachable(permission)
  }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.29.0` |
| `@typescript-eslint/parser`        | `2.29.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.9` |
| `node`                             | `12.12.0` |
| `npm`                              | `6.13.6` |

**Comment by bradzacher** [comment]
the rule doesn't yet check to see if the containing function has the `async` keyword.
Shouldn't be too difficult of a change.
Happy to accept a PR.

**Comment by yinm** [comment]
Can I try it?
